src/PostgREST/QueryBuilder.hs

```diff
@@ -61,9 +61,22 @@ import           Data.Scientific         ( FPFormat (..)
 import           Prelude hiding          (unwords)
 import           PostgREST.ApiRequest    (PreferRepresentation (..))
 
+{-| The generic query result format used by API responses. The location header
+    is represented as a list of strings containing variable bindings like
+    @"k1=eq.42"@. If unused, it's null/Nothing rather than the empty list
+    because 'PostgreSQL.Binary.Decoder.arrayDimension' cannot decode an empty
+    array!
```

Comment:
@nikita-volkov perfect!

@league I'll make a PR to update all deps to their newest version including hasql stuff. Once that's merged could you send a PR to refactor this part of the code?